experiments/data/prepare_hendrycksSTEM_dataset.py

```diff
@@ -32,24 +32,33 @@
 NAME = "hendrycks_STEM"
 
 
-def make_test_dataset(tasks):
+def make_train_dataset(tasks, data_split):
     task_dict = lm_eval.tasks.get_task_dict(tasks)
-    task_dict_items = [
-        (name, task) for name, task in task_dict.items() if task.has_test_docs()
-    ]
 
     docs = []
-    for _, task in task_dict_items:
-        task_doc_func = task.test_docs
+    task_sizes = {}
```

Comment:
Not necessary to do now and this might just be my hatred of `if ... elif ... elif` statements but to allow the user to pass any split, you could change this to be

```python
for task_name, task in task_dict.items():
   task_cond = f"has_{data_split}_docs"
   task_func = f"{data_split}_docs"
   if getattr(tash, task_cond):
      task_doc_func = getattr(tash, task_cond)
   else:
      raise ValueError("task must have training, validation or test split")
```